### ISSUE TITLE
"Request is missing Authentication Token" after upgrade to 3.25.0

### ISSUE THREAD

**Issue by andrmoel** [serverless/serverless#11561]
### Are you certain it's a bug?

- [X] Yes, it looks like a bug

### Is the issue caused by a plugin?

- [X] It is not a plugin issue

### Are you using the latest v3 release?

- [X] Yes, I'm using the latest v3 release

### Is there an existing issue for this?

- [X] I have searched existing issues, it hasn't been reported yet

### Issue description

After I have upgraded from 2.24.1 to 3.25.0 I do get the following error while I deploy:

> "Request is missing Authentication Token"

### Service configuration (serverless.yml) content

```yaml
N/A
```

### Command name and used flags

serverless deploy --stage=staging

### Command output

```shell
yarn run v1.22.19
$ cd microservices/core && serverless deploy --verbose

Deploying portal to stage dev (eu-central-1)

Packaging
Using configuration:
{
  "webpackConfig": "./../../webpack.config.js",
  "includeModules": false,
  "packager": "yarn",
  "packagerOptions": {},
  "keepOutputDirectory": false,
  "concurrency": 8
}
Removing /home/test/Projekte/test-project/.webpack
[Webpack] Building with Webpack
[Webpack] Packaging modules
Zip service: /home/test/Projekte/test-project/.webpack/service [324 ms]
[Webpack] Copying existing artifacts
Setting artifact for function 'info' to '.serverless/portal.zip'
Setting artifact for function 'version' to '.serverless/portal.zip'
Setting artifact for function 'onPreSignUp' to '.serverless/portal.zip'
Setting artifact for function 'onPostConfirmation' to '.serverless/portal.zip'
Setting artifact for function 'onCustomMessage' to '.serverless/portal.zip'
Remove /home/test/Projekte/test-project/.webpack
Generating custom CloudFormation resources
Removing /home/test/Projekte/test-project/.webpack done
Retrieving CloudFormation stack

✖ Stack portal-dev failed to deploy (32s)
Environment: linux, node 14.19.0, framework 3.25.0 (local), plugin 6.2.2, SDK 4.3.2
Credentials: Local, "default" profile
Docs:        docs.serverless.com
Support:     forum.serverless.com
Bugs:        github.com/serverless/serverless/issues

Error:
Request is missing Authentication Token

1 deprecation found: run 'serverless doctor' for more details
error Command failed with exit code 1.
```

### Environment information

```shell
Running "serverless" from node_modules
Framework Core: 3.25.0 (local) 3.1.1 (global)
Plugin: 6.2.2
SDK: 4.3.2
```

**Comment by sbmvirdi** [comment]
Hi even i am also facing the same issue while using serverless.
It was working fine few days back.

**Comment by modanwalrahul31** [comment]
Hi, even I am also facing the same issue while using serverless.
From Today.

**Comment by JunBeum-Cho** [comment]
I am also getting the same issue ughhhhh spent whole day for this

**Comment by modanwalrahul31** [comment]
I don't think this issue comes after the upgrade, even existing projects are still working with the upgraded.

**Comment by andrmoel** [comment]
> I don't think this issue comes after the upgrade, even existing projects are still working with the upgraded.

I tested it with 2.24.1 -> works as expected
Only 2.25.0 is facing that issue

**Comment by luksfarris** [comment]
I tested with several versions of serverless, from 1.61 to 3.21, all have this error.
I don't think it's related to the version

**Comment by nikhiltechahead** [comment]
I'm getting the same error when deploying the code to AWS lambda from today only. I've a serverless version **3.25.0**.

**Comment by sbmvirdi** [comment]
yes even i tried using older versions but nothing seems to work. 
I have changed my whole pipelines so many times.

**Comment by AnishLushte07** [comment]
#https://github.com/serverless/serverless/discussions/11559

**Comment by nikhiltechahead** [comment]
Something is really broken here unexpectedly. Spent almost all day checking what I'm doing wrong.

**Comment by sathishkumar16** [comment]
https://github.com/serverless/serverless/issues/11560

**Comment by gskim** [comment]
i'm waiting....

**Comment by coder-pikachu** [comment]
We are also facing the same issue. The build was working until yesterday...

**Comment by daum1820** [comment]
Facing the same problem but I've also update "aws-sdk": "2.1262.0" which are also not working after downgrading the serveless againg to 3.24.1

**Comment by sbmvirdi** [comment]
but what i am thinking is how it was working until yesterday because 3.25.0 release was merged 7 days ago

**Comment by daum1820** [comment]
https://github.com/aws/aws-sdk-js/issues/4286

**Comment by luksfarris** [comment]
Yeah, had to `npm install --save-dev aws-sdk@2.1259.0`. Thanks guys

**Comment by andrmoel** [comment]
> Facing the same problem but I've also update "aws-sdk": "2.1262.0" which are also not working after downgrading the serveless againg to 3.24.1

@daum1820 This could be the issue. Serverless has a dependency to aws-sdk.

As I have written before. Version 3.24.1 is working for us. Please don't simply downgrade your package.json because it does not downgrade the aws-sdk dependency. You need to do a clean install!

**Comment by sbmvirdi** [comment]
yes serverless has bumped new version of aws js library and upgraded it. the serverless needs to downgrade the version back to old to make it working

**Comment by coder-pikachu** [comment]
> > Facing the same problem but I've also update "aws-sdk": "2.1262.0" which are also not working after downgrading the serveless againg to 3.24.1
> 
> @daum1820 This could be the issue. Serverless has a dependency to aws-sdk.
> 
> As I have written before. Version 3.24.1 is working for us. Please don't simply downgrade your package.json because it does not downgrade the aws-sdk dependency. You need to do a clean install!

Clean install meaning?

**Comment by andrmoel** [comment]
> Clean install meaning?

```
yarn remove serverless
yarn add serverless@version
```

Be aware: If you have `aws-sdk` as dependency in your package.json, you also need to remove this one otherwise it won't get downgraded.

**Comment by sbmvirdi** [comment]
@andrmoel i am using serverless deploy in pipelines so it should not fetch the latest version for aws sdk js right ?

**Comment by AnishLushte07** [comment]
Deleting node modules worked for me

**Comment by andrmoel** [comment]
@sbmvirdi depends on your package.json and yarn.lock (Sorry, we only use yarn. I think for npm it is package-lock.json).

Your yarn.lock contains the state of truth and should be checked in version control. Just check if your yarn.lock contains the correct version of `serverless` and `aws-sdk`. Then you should be green.

**Comment by pankaj-torinit** [comment]
I am using serverless for laravel deployment using bref package. Getting the same error. Any solution?

**Comment by ziamler** [comment]
I'm also facing the same issue since morning. I have tried lots of solutions but don’t work. at last, I have tried to remove node_modules and use the node 16 version. it's working fine for me.

**Comment by sbmvirdi** [comment]
@andrmoel but i am not pushing my lock file to github i am just pushing my package json file to repo and doing npm i in the ci. i tried using specific version for serverless in package json  (1.23.0) but it was getting failed.

**Comment by andrmoel** [comment]
@sbmvirdi then you should also push your lock. It is best practice because you always want to have the same build on your machine that you have in production. In your case, the fix and minor versions differ. This can cause side effects.

But we are getting off-topic ;)

**Comment by brainframe-me** [comment]
Same issue here, with v2 of serverless. Even tried upgrading from 2.72.2 to 2.72.3 but does not change anything.
So a build that worked perfectly last week is now failing with "Request is missing Authentication Token".

Your Environment Information ---------------------------
     Operating System:          linux
     Node Version:              16.13.1
     Framework Version:         2.72.3
     Plugin Version:            5.5.4
     SDK Version:               4.3.2
     Components Version:        3.18.2

**Comment by sbmvirdi** [comment]
@andrmoel in github actions it doesn't matter actually. i just tried installing the serverless 3.24.1 in my runner and it just failed. i don't understand how its failing for older version. actions are not using the package-json.

**Comment by coder-pikachu** [comment]
We are using Code Build and its failing there too with the following versions -
Node: 16
serverless: 3.24.1/3.24.0

**Comment by coder-pikachu** [comment]
> @andrmoel in github actions it doesn't matter actually. i just tried installing the serverless 3.24.1 in my runner and it just failed. i don't understand how its failing for older version. actions are not using the package-json.

Same here with Code Build. It recreates the environment hence references to older builds should not cause issues "ideally". But here the behaviour is really strange.

**Comment by andrmoel** [comment]
@sbmvirdi @coder-pikachu please check your yarn.lock file.

You should find:

```yaml
...

serverless@^3.24.1:
  version "3.24.1"
  resolved "https://registry.yarnpkg.com/serverless/-/serverless-3.24.1.tgz#2d361e97192d4565131e8b11e90dc033d2d1a74f"
  
...

aws-sdk@^2.1247.0, aws-sdk@^2.814.0:
  version "2.1253.0" # !!!!!!!! THIS MUST NOT BE 2.1262.0 !!!!!!!!
  resolved "https://registry.yarnpkg.com/aws-sdk/-/aws-sdk-2.1253.0.tgz#28ac12775f4d1589e366db6f2b162d8ea150c2ad"

...
```

**Comment by coder-pikachu** [comment]
We are using npm for managing the packages.

**Comment by coder-pikachu** [comment]
> @sbmvirdi @coder-pikachu please check your yarn.lock file.
> 
> You should find:
> 
> ```yaml
> ...
> 
> serverless@^3.24.1:
>   version "3.24.1" # > > > > THIS MUST BE 3.24.1 < < < <
>   resolved "https://registry.yarnpkg.com/serverless/-/serverless-3.24.1.tgz#2d361e97192d4565131e8b11e90dc033d2d1a74f"
>   
> ...
> 
> aws-sdk@^2.1247.0, aws-sdk@^2.814.0:
>   version "2.1253.0" # > > > > THIS MUST NOT BE 2.1262.0 < < < <
>   resolved "https://registry.yarnpkg.com/aws-sdk/-/aws-sdk-2.1253.0.tgz#28ac12775f4d1589e366db6f2b162d8ea150c2ad"
> 
> ...
> ```

We are using npm for managing the packages

**Comment by andrmoel** [comment]
The idea is the same. Just ensure the correct versions of serverless and aws-sdk are installed.

I know... dependencies are a b***

**Comment by krkcode60** [comment]
I experience the same error and failure of deployments. Everything used to work up until this morning and untouched modules started breaking.

**Comment by noglitchyo** [comment]
> Same issue here. Using the following in CircleCI : `yarn global add serverless@3.17`. There is no package.json with our application, it is a PHP application, we only use Node during deployment.
> 
> I tried different Serverless Framework version (3.24, 3.21, etc) and got the same results. I also tried to force the `aws-sdk` version without success.
> 
> ```
> Environment: linux, node 16.18.1, framework 3.17.0, plugin 6.2.2, SDK 4.3.2
> ```
> 
> Also tried with node 14, same issue.

I finally switched from global installation to a package.json installation.

My package.json is the following

```json
{
  "name": "application",
  "version": "1.0.0",
  "dependencies": {
    "serverless": "3.17.0",
    "aws-sdk": "2.1243.0"
  },
  "resolutions": {
    "serverless/aws-sdk": "2.1243.0"
  }
}
```
and the pipeline script looks like this:
```bash
yarn --frozen-lockfile
yarn run sls deploy --verbose --stage mystage
```
With this config, it works.

**Comment by benoitvallon** [comment]
As the issue is only with `aws-sdk`, you can add the following resolution to your `package.json` and force your `aws-sdk` version to 2.1259.0 which seems to be the last one working. This way you can keep `serverless` to the last version `^3.25.0`
```
"dependencies": {
  ...
  "aws-sdk": "2.1259.0"
  ...
},
"resolutions": {
  "aws-sdk": "2.1259.0"
},
  ```

**Comment by coder-pikachu** [comment]
> > Same issue here. Using the following in CircleCI : `yarn global add serverless@3.17`. There is no package.json with our application, it is a PHP application, we only use Node during deployment.
> > I tried different Serverless Framework version (3.24, 3.21, etc) and got the same results. I also tried to force the `aws-sdk` version without success.
> > ```
> > Environment: linux, node 16.18.1, framework 3.17.0, plugin 6.2.2, SDK 4.3.2
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Also tried with node 14, same issue.
> 
> I finally switched from global installation to a package.json installation.
> 
> My package.json is the following
> 
> ```json
> {
>   "name": "application",
>   "version": "1.0.0",
>   "dependencies": {
>     "serverless": "3.17.0",
>     "aws-sdk": "2.1243.0"
>   },
>   "resolutions": {
>     "serverless/aws-sdk": "2.1243.0"
>   }
> }
> ```
> 
> and the pipeline script looks like this:
> 
> ```shell
> yarn --frozen-lockfile
> yarn run sls deploy --verbose --stage mystage
> ```
> 
> We are using `yarn` to manage dependencies. The `package.json` and `yarn.lock` files are versionned. With this config, it works.

This worked for us! Thank you!

**Comment by nikhiltechahead** [comment]
> As the issue is only with `aws-sdk`, you can add the following resolution to your `package.json` and force your `aws-sdk` version to 2.1259.0 which seems to be the last one working. This way you can keep `serverless` to the last version `^3.25.0`
> 
> ```
> "dependencies": {
>   ...
>   "aws-sdk": "2.1259.0"
>   ...
> },
> "resolutions": {
>   "aws-sdk": "2.1259.0"
> },
> ```

This resolved the issues. Thanks.

**Comment by medikoo** [comment]
It's a regression in AWS-SDK. To increase visibility please comment (and +1) https://github.com/aws/aws-sdk-js/issues/4286 and https://github.com/aws/aws-sdk-js/issues/4285

**Comment by MehranJanfeshan** [comment]
@medikoo why did you close it? There is no clear solution yet; even downgrading doesn’t work.

**Comment by brainframe-me** [comment]
I finally got it to work from my side. Make sure to understand if you are using serverless npm installed globally or as a local project in your packages. When your serverless is installed through npm install -global, you can also fix the correct aws-sdk as following:

npm i -g serverless@2.72.2 aws-sdk@2.1259.0 (tested and finally working)
or
npm i -g serverless@3.25.0 aws-sdk@2.1259.0 (not tested, but assuming this also fixes the issue here)

**Comment by MehranJanfeshan** [comment]
> I finally got it to work from my side. Make sure to understand if you are using serverless npm installed globally or as a local project in your packages. When your serverless is installed through npm install -global, you can also fix the correct aws-sdk as following:
> 
> npm i -g serverless@2.72.2 aws-sdk@2.1259.0 (tested and finally working) or npm i -g serverless@3.25.0 aws-sdk@2.1259.0 (not tested, but assuming this also fixes the issue here)

this worked very well, thanks

**Comment by medikoo** [comment]
> @medikoo why did you close it? There is no clear solution yet; even downgrading doesn’t work.

@MehranJanfeshan I didn't close it (it's other duplicate issue that was closed)

**Comment by RohitSaini15** [comment]
facing the same issue while deploying the aws lambda after using the serverless-python-requirements plugin. Without it, it 's working fine

**Comment by trivikr** [comment]
This is maintainer of AWS SDK for JavaScript, and I'm visiting from https://github.com/aws/aws-sdk-js/issues/4286

If you can provide a minimal repro there, it would be helpful for us to triage, debug and provide a fix.

**Comment by medikoo** [comment]
Just released v3.25.1 version of the Framework, which fixes AWS SDK to the last working version. Please upgrade and the issue should be fixed

**Comment by trivikr** [comment]
Commits between https://github.com/serverless/serverless/compare/v3.25.0...v3.25.1

This was fixed by pinning aws-sdk to `2.1259.0` in https://github.com/serverless/serverless/commit/cb810854a619429b8ec2a4c1dbbd77bb273d015d right?

**Comment by trivikr** [comment]
Do you suspect the issue was introduced in aws-sdk `2.1262.0` or `2.1260.0`?
Why do the recommendations pin `aws-sdk` version to `2.1259.0`?

Can someone pin `aws-sdk` to `2.1261.0`, and check if it works for you?

Even better, if you can provide minimal repro in https://github.com/aws/aws-sdk-js/issues/4286, so that we can compare it against different versions of aws-sdk.

**Comment by andrmoel** [comment]
@trivikr I have setup a simple project to reproduce the issue

https://github.com/andrmoel/serverless-aws-bug

**Comment by trivikr** [comment]
> I have setup a simple project to reproduce the issue
> 
> [andrmoel/serverless-aws-bug](https://github.com/andrmoel/serverless-aws-bug)

Thank you @andrmoel 

Do you have another repro without serverless?
i.e. a repro which just has dependency on `aws-sdk` in package.json and nothing else.

**Comment by andrmoel** [comment]
@trivikr sorry can't provide this.
But I have created also a working example using the old version numbers.
Just checkout the branch "working"

https://github.com/andrmoel/serverless-aws-bug/tree/working

**Comment by andrmoel** [comment]
@trivikr sorry. May there is a misunderstanding here.
The aws-sdk is a dependency of serverless. So it is part of the project

**Comment by trivikr** [comment]
> The aws-sdk is a dependency of serverless. So it is part of the project

AWS SDK for JavaScript is maintained by AWS, but serverless is community maintained.

Looking at the current repro and fixes, the issue is likely with `aws-sdk` introduced in `v2.1262.0`.
We just need a repro to triage, debug and fix the issue in https://github.com/aws/aws-sdk-js/issues/4286

**Comment by NGIamarino** [comment]
> I finally got it to work from my side. Make sure to understand if you are using serverless npm installed globally or as a local project in your packages. When your serverless is installed through npm install -global, you can also fix the correct aws-sdk as following:
> 
> npm i -g serverless@2.72.2 aws-sdk@2.1259.0 (tested and finally working) or npm i -g serverless@3.25.0 aws-sdk@2.1259.0 (not tested, but assuming this also fixes the issue here)

This worked for me.
Thank you!

**Comment by trivikr** [comment]
> We have a fix available in [aws/aws-sdk-js#4288](https://github.com/aws/aws-sdk-js/pull/4288) using the repro which @andrmoel provided.

Can you merge bump to yarn modern in https://github.com/andrmoel/serverless-aws-bug/pull/1?

I'll post a draft PR using [yarn patch](https://yarnpkg.com/cli/patch) to verify our fix prior to release.

**Comment by trivikr** [comment]
Verified that fix from https://github.com/aws/aws-sdk-js/pull/4288 fixes 
"Request is missing Authentication Token" issue.

Refer Pull request in https://github.com/trivikr/serverless-aws-bug/pull/1
We're aiming to release this fix in `v2.1263.0` within the next 24 hours.

**Comment by medikoo** [comment]
@trivikr great thanks for fast action! 💯 🎉

**Comment by trivikr** [comment]
Release `aws-sdk@v2.1263.0` is out, and I verified that "Request is missing Authentication Token" error is no longer thrown.

https://github.com/andrmoel/serverless-aws-bug/pull/2

@medikoo Release v3.25.2 after reverting commit https://github.com/serverless/serverless/commit/cb810854a619429b8ec2a4c1dbbd77bb273d015d?

**Comment by han-fs** [comment]
I added serverless config and it works.
```
serverless config credentials --provider aws --key ... --secret ...
```

**Comment by andrmoel** [comment]
@trivikr thank you for the quick response

**Comment by MehranJanfeshan** [comment]
> I added serverless config and it works.
> 
> ```
> serverless config credentials --provider aws --key ... --secret ...
> ```
> 
> serverless version: 2.72.3, 3.25.1 aws sdk version: 2.x

@han-fs doesn’t work for pipelines that use assumed roles, and exposing keys and secrets in pipelines is not ideal.

**Comment by Scoppio** [comment]
AWS-SDK has released a new version which fix the issue.

**Comment by medikoo** [comment]
> @medikoo Release v3.25.2 after reverting commit cb81085?

@trivikr yes, with the next version, we'll get back to the regular versioning of AWS-SDK

**Comment by BasilaryGroup** [comment]
**Your fix works for us**. We are now able to deploy everything that failed yesterday.

Thank you!

**Comment by cforce** [comment]
I getting crzy , i tried dozens of version combinbations proposed here ..but non work.
The string things is thate it alway complains with the same  error message

```
>sls deploy --aws-profile dev    
Running "serverless" from node_modules
Using serverless-localstack
Environment: win32, node 19.2.0, framework 3.25.1 (local) 3.25.1v (global), plugin 6.2.2, SDK 4.3.2
Docs:        docs.serverless.com
Support:     forum.serverless.com
Bugs:        github.com/serverless/serverless/issues

Error:
Cannot run local installation of the Serverless Framework by the outdated global version.
Please upgrade via:

npm install -g serverless
Note: Latest release can run any version of the locally installed Serverless Framework.

Alternatively run locally installed version directly via:

npx serverless <command> <options>
```
```
{
  "dependencies": {
    "serverless": "^3.25.1",
      "aws-sdk": "2.1259.0"
      },
  "devDependencies": {
    "serverless-deployment-bucket": "^1.6.0",
    "serverless-localstack": "^1.0.1",
    "serverless-offline": "^11.6.0",
    "serverless-plugin-datadog": "^5.12.0"
  }
}
```

**Comment by enchorb** [comment]
+1 on what @MehranJanfeshan said, getting `The security token included in the request is invalid.` for our CI pipeline which is assuming an IAM role through OIDC. The credentials are correct because other `aws-sdk` commands are working fine.
